pybr2021/cogs/schedules.py

```diff
@@ -0,0 +1,116 @@
+import asyncio
+import datetime
+from datetime import datetime, timedelta
+from pprint import pprint
+
+import bot_msg
+import httpx
+from decouple import config
+from discord.ext import commands, tasks
+from loguru import logger
+from pytz import timezone
+
+CALENDAR_URL='https://www.googleapis.com/calendar/v3/calendars/7siodq5un9gqbqd4mmgf2poiqs@group.calendar.google.com/events?key=AIzaSyAIn8DyZFtthupLozgwIX3NUURFMWEIPb4&timeMin=2021-10-11T00:00:00.000Z&timeMax=2021-10-18T00:00:00.000Z&singleEvents=true&maxResults=9999&timeZone=UTC'
+CALENDER_TIMEZONE= 'UTC'
+SHOW_TIMEZONE='America/Sao_Paulo'
+DISCORD_MSG_CHANNEL_ID='859819206584959007'  # Python Brasil 2021 > Geral
+DATE_FMT = "%d/%m/%Y %H:%M:%S"
+HOUR_FMT = "%H:%M"
+class Schedules(commands.Cog):
+
+    def  __init__(self,bot):
+        self.loop = asyncio.get_event_loop()
+        self._events = None
+        self.index = {}
+        self._bot = bot
+        self.alerts_type=["talk","closing","keynote","panel"]
+        self.load_events.start()
+
+    @tasks.loop(minutes=1)
+    async def load_events(self):
+        logger.info("Loading calendar events.")
+        url = CALENDAR_URL
+        semaphore = asyncio.BoundedSemaphore(10)
+        async with httpx.AsyncClient() as client:
+            response = await self.http_get_json(semaphore, client, url)
+        
+        logger.info("Parsing events")
+        self._events = await self.parse_events(response)
+        self.index = self.create_index(self._events)
+        logger.info("Calendar finished load")
+
+    async def parse_events(self,response):
+        events=[]
+        for item in response.get("items"):
+            if item.get('extendedProperties').get('private').get('type') in self.alerts_type:
+                events.append(
+                    {
+                        'start': datetime.strptime(item.get('start').get('dateTime'),'%Y-%m-%dT%H:%M:%SZ').replace(tzinfo=timezone(CALENDER_TIMEZONE)),
+                        'timezone':item.get('start').get('timeZone'),
+                        'location':item.get('location'),
+                        'title':item.get('extendedProperties').get('private').get('title'),
+                        'author':item.get('extendedProperties').get('private').get('author'),
+                        'discord_channel':item.get('extendedProperties').get('private').get('discord_channel'),
+                        'type':item.get('extendedProperties').get('private').get('type'),
+                        'youtube_channel':item.get('extendedProperties').get('private').get('youtube_channel'),  
+                    }
+            )
+        return events
+
+    def create_index(self,events):
+        index = {}
+        for event in events:
+            index[event["start"].date()] = []
```

Comment:
```suggestion
        index = defaultdict(list)
```
`defaultdict` é tão legal! :)
```
from collections import defaultdict
```